Generating all subsets iteratively

dhruvinfo28 opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests]

Yes

### Description

closes #3067. Algorithm for generating all subsets iteratively added with time and space complexity.

### Checklist

- [ Yes] I've read the contribution guidelines.
- [ Yes] I've checked the issue list before deciding what to submit.
- [ Yes] I've edited the `README.md` and link to my code.

dhruvinfo28 committed:
Generating all subsets iteratively

HarshCasper reviewed (changes requested):
Few changes: 
- Push the primary logic of code in some other function
- Use the `main()` function only for driver code
- Format and indent the Code properly 
- Use multi-line comments wherever possible

dhruvinfo28 committed:
Requested changes done

Driver code in main
Primary logic of code in some other function
Multiline comments
Indentation and code formatted

dhruvinfo28 commented:
@HarshCasper  I made the requested changes kindly review.

dhruvinfo28 committed:
Merge branch 'master' into forIssue3067

dhruvinfo28 commented:
/assign
